### DISCUSSION TITLE
How to create typebox type that is the same as typescript `typeof MyEnum`?

### DISCUSSION THREAD

**Discussion by meightythree** [sinclairzx81/typebox#509] [Q&A] [answered]
I have the following enum:
```
enum MyEnum {
   first = 'first',
   second = 'second',
}
```
How to create a typebox type which represents the enum content? 
In typescript it is done by `typeof MyEnum`

**Comment by sinclairzx81** [comment, marked as answer]
@meightythree Hi,

You can pass a `enum` into `Type.Enum` which will create a union (anyOf) schema. You can then use `Static<typeof T>` which will return the original enum type (as it would be understood by TypeScript)

```typescript
enum MyEnum {
   first = 'first',
   second = 'second',
}

const T = Type.Enum(MyEnum)    // const T = {
                               //   anyOf: [{
                               //       type: 'string',
                               //       const: 'first',
                               //   }, {
                               //       type: 'string',
                               //       const: 'second',
                               //     }
                               //   ],
                               // }

type T = Static<typeof T>      // type T = MyEnum 

function test(value: T) {}   

test(MyEnum.first)           // ok - matches anyOf variant
test(MyEnum.second)          // ok - matches anyOf variant
test(MyEnum)                 // fail - not a anyOf variant
```
In the above, take note of the third test case. Note that because TypeBox encodes `enum` as `union` (in which it only encodes the enum variants), information about the original `MyEnum` keys are lost (as these may be different). Because of this, there is no way to reproduce the original `typeof MyEnum` in full, so this is generally unsupported.

Hope this helps
S

**Reply by meightythree** [reply]
Thank you for the detailed answer!
